oidc-controller/api/verificationConfigs/helpers.py

```diff
@@ -0,0 +1,43 @@
+from .variableSubstitutions import variable_substitution_map
+
+
+class VariableSubstitutionError(Exception):
+    """Custom exception for if a variable is used that does not exist."""
+
+    def __init__(self, variable_name: str):
+        self.variable_name = variable_name
+        super().__init__(f"Variable '{variable_name}' not found in substitution map.")
+
+
+def replace_proof_variables(proof_req_dict: dict) -> dict:
+    """
+    Recursively replaces variables in the proof request with actual values.
+    The map is provided by imported variable_substitution_map.
+    Additional variables can be added to the map in the variableSubstitutions.py file,
+    or other dynamic functionality.
+
+    Args:
+        proof_req_dict (dict): The proof request dictionary from the resolved config.
+
+    Returns:
+        dict: The updated proof request dictionary with placeholder variables replaced.
+    """
+
+    for k, v in proof_req_dict.items():
+        # If the value is a dictionary, recurse
+        if isinstance(v, dict):
+            replace_proof_variables(v)
+        # If the value is a list, iterate trhough list items and recurse
```

Comment:
small spelling mistake here.

Reply:
Thx, will throw a fix for that in a subsequent dependency update to not have to re-request review here 😄